contracts/pools/weighted/WeightedPool2Tokens.sol

```diff
@@ -185,20 +188,29 @@ contract WeightedPool2Tokens is
         emit SwapFeePercentageChanged(swapFeePercentage);
     }
 
-    function _isOracleEnabled() private view returns (bool) {
-        return _miscData.decodeBool(_MISC_ORACLE_ENABLED_OFFSET);
+    function enableOracle() external authenticate {
+        _setOracleEnabled(true);
```

Comment:
```suggestion
        _setOracleEnabled(true);
        _cacheInvariantAndSupply();
```

Reply:
Agreed as we discussed off-line... let's add some tests for it